XRTK-Core/Packages/com.xrtk.core/Runtime/Utilities/RenderPipelineUtilities.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) XRTK. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine.Rendering;
+using XRTK.Extensions;
+using RenderPipeline = XRTK.Definitions.Utilities.RenderPipeline;
+
+namespace XRTK.Utilities
+{
+    public static class RenderPipelineUtilities
+    {
+        private const string urpAssetTypeName = "UniversalRenderPipelineAsset";
+        private const string hdrpAssetTypeName = "HDRenderPipelineAsset";
+
+        /// <summary>
+        /// Gets the <see cref="RenderPipeline"/> currently in use and active.
+        /// </summary>
+        /// <returns></returns>
+        public static RenderPipeline GetActiveRenderingPipeline()
+        {
+            var renderPipelineAsset = GraphicsSettings.renderPipelineAsset;
+            if (renderPipelineAsset.IsNull())
+            {
+                return RenderPipeline.Legacy;
+            }
+
+            switch (renderPipelineAsset.GetType().Name)
+            {
+                case urpAssetTypeName:
```

Comment:
Wondering if we can use pattern matching for the types instead of using the string name

Reply:
The actual types are only available if the URP / HDRP package is installed and this check needs to work even when no render pipeline package is in use. So string is all we got.

Reply:
@FejZa do you know if those packages define symbols for the features?

Reply:
No they don't. Freya Holmér put a lot of thoughts into this and this is pretty much what she came up with in the end. It's not perfect but I don't think currently there is any better way to check this.